### ISSUE TITLE
FCLK values not reading correctly

### ISSUE THREAD

**Issue by BeatPlay** [irusanov/ZenTimings#1]
Just downloaded 1.1.0 and the values for FCLK are absolutely random. Reads anywhere between 2500 and 2800 while UCLK and MCLK are correctly 1900.

Asus ROG Strix X570i + 3950X

**Comment by irusanov** [comment]
That's what is reported on Zen2 when SOC/Uncore OC mode is not used. It should also drop to lower frequencies when the PC is not in use. If you force the OC mode or disable DF C-States it should read constant ~1900.

It's hard to figure everything out with all that combinations of bios versions, motherboards and CPUs, so if possible, can you run a Debug Report and attach it here.

**Comment by BeatPlay** [comment]
> It should also drop to lower frequencies when the PC is not in use.

Those weird values were actualy during Idle
I also think we have to rename the issue to "values not loading at all", because it now refuses to work completely with the updated BIOS of my Board.

[Debug_Report_26638098,441356.txt](https://github.com/irusanov/ZenTimings/files/5119067/Debug_Report_26638098.441356.txt)

**Comment by irusanov** [comment]
There's probably yet another change in new AGESA that I have to figure out in order to support it.
Your DRAM base address is 0, so it can't get the power table. It might be possibly a bug in that exact BIOS version as well.
Unless it is some obscure bug in ZenTimings, then I assume Hwinfo can't read the values either. Even RyzenMaster might fail to read them.

Please try with those apps if you have them installed and report back, but it seems I will have to buy a new motherboard to continue with the development. Still using my X370-based Crosshair VI Hero.

**Comment by BeatPlay** [comment]
Ryzen Master and HWiNFO work as expected.
In case this helps, this is from the changelog of the BIOS:

> Update AM4 AGESA combo V2 PI 1.0.8.0

**Comment by irusanov** [comment]
Ok, thanks. Last thing to try. After opening hwinfo at least once, can you open ZenTimings again and see if it works?
I see that smu version is also 00.00.00, which means the SMU is not responding at all for some reason.

**Comment by BeatPlay** [comment]
Thats a thing I kind of did while checking, had Ryzen Master and HWiNFO open and then opened ZenTimings besides that to compare the values that did load.

Also the SMU thing is weird, if Ryzen Master works there shouldnt be a reason that it shouldnt respond to the same requests form a different program right?

**Comment by irusanov** [comment]
There might be plenty of reasons, the main one is that I don't have access to any non-public documentation.
Something might have changed along the way with the new combo V2 bioses, which I haven't discovered yet and will probably have to buy such a motherboard in order to reproduce it and debug it.

Again, thanks for the report.

**Comment by BeatPlay** [comment]
No Problem, I guess we can close this then.

Probably impractical, but if you want I'd be down to collect some more data on this if you point me in the right directions.

**Comment by irusanov** [comment]
It's a valid issue, so I'm leaving it open.
I guess you can try my SMU Debug Tool, but it's not very user-friendly and can be dangerous in certain cases when things are tested blindly: https://github.com/irusanov/SMUDebugTool

Two simple tests that are harmless is to open the tool, select SMU, then click on Send.
This sends a test command in order to see if the SMU responds. Should print 0x1 if ok.
Next is to change command to 0x2 (or just 2)  which is "get SMU version", which also should print a value and say OK.
Both these are safe to try.

Last thing is to click Scan, but there's a chance that will lock the system up and will need a full power cycle, not just a restart.

All this needs to be done after a fresh boot without anything other monitoring programs opened and running (like RyzenMaster and hwinfo).

**Comment by BeatPlay** [comment]
0x1 outputs this:
```
HEX: 0x1
INT: 1

Detected NUMA nodes. (1)

OK
```
0x2 outputs this:
```
HEX: 0x2E3F00
INT: 3030784

HEX: 0x1
INT: 1

Detected NUMA nodes. (1)

OK
```
This is the scan result:
```
CMD:  0x3B10524
RSP:  0x3B10570
ARG:  0x3B10A40

CMD:  0x3B10528
RSP:  0x3B10574
ARG:  0x3B10A60

CMD:  0x3B10530
RSP:  0x3B1057C
ARG:  0x3B109C4
```

**Comment by irusanov** [comment]
Hmm, so there isn't a reason that it should not work. Maybe a conflict with the other tools.
Try ZenTimings after a fresh boot prior to opening any other tools, maybe even close it and launch it a second time if the first one does not give you the expected result.

**Comment by BeatPlay** [comment]
The first thing I tried was rebooting, but now its back to working in all combinations with different software I had. Weird, but atleast something. Just those weird values now, but you explained that up there.

**Comment by irusanov** [comment]
Should be resolved with v1.2.0
